### PR TITLE
fix: [MSSQL] object reassignment when binding is deleted

### PR DESCRIPTION
**Pull request by zucchinidev**
[#185100152](https://www.pivotaltracker.com/story/show/185100152)

### Checklist:

* [ ] Have you added Release Notes in the docs repositories?
* [x] Have you followed the [Conventional Commits specification](https://www.conventionalcommits.org/en/v1.0.0/#summary)?

### CODE REVIEW

File: providers/terraform-provider-csbsqlserver/connector/connector.go [hunk 1]
```diff
@@ -5,35 +5,55 @@ import (
 	"context"
 	"database/sql"
 	"fmt"
-	"net/url"
 
-	_ "github.com/denisenkom/go-mssqldb"
+	_ "github.com/microsoft/go-mssqldb"
```
Comment (fnaranjo-vmw) [inline providers/terraform-provider-csbsqlserver/connector/connector.go:9]: 🎉

File: providers/terraform-provider-csbsqlserver/go.mod [hunk 2]
```diff
@@ -0,0 +1,74 @@
+module github.com/cloudfoundry/csb-brokerpak-azure/terraform-provider-csbsqlserver
```
Comment (fnaranjo-vmw) [inline providers/terraform-provider-csbsqlserver/go.mod:1]: Just curios.
Was converting this folder into a full-fledged Go module really needed or just an opportunistic improvement?

**Comment by fnaranjo-vmw** [comment]
I would have loved this PR to be the Minimum Viable Change to fix the issue.
Instead, it looks like a bunch of very nice improvements, with only some of them contributing to the actual fix.

This makes it very difficult to reason about the changes, and look for possible mistakes, typos, etc.
Since it is impossible to distinguish:
- Aesthetic improvements
- Usability improvements
- Strictly required changes to implement the fix

Looking at the commit history doesn't help either, since almost of these changes were introduced in the first commit.

**Comment by pivotal-marcela-campo** [comment]
@fnaranjo-vmw I understand your points. It is valuable to understand how this PR was generated. The original idea was to extract this code into a common provider in its own repository, not retrofit the changes in this one. We backtracked on that decision and splitting the commits would have been as hard as reviewing the changes.

File: providers/terraform-provider-csbsqlserver/connector/connector.go [hunk 3]
```diff
@@ -72,14 +97,155 @@ func (c *Connector) ReadBinding(ctx context.Context, username string) (result bo
 	})
 }
 
+func (c *Connector) CheckDatabaseExists(ctx context.Context, dbName string) (result bool, err error) {
+	return result, c.withDefaultDBConnection(func(db *sql.DB) (err error) {
+		statement := `SELECT 1 FROM master.dbo.sysdatabases where name=@p1`
```
[the remark this answers is not in the document]
Comment (pivotal-marcela-campo) [reply providers/terraform-provider-csbsqlserver/connector/connector.go]: Changed the query to use a system view, this works fine in Azure and AWS, tried all engine versions

**Review by pivotal-marcela-campo — APPROVED** [review]

File: providers/terraform-provider-csbsqlserver/connector/connector.go [hunk 4]
```diff
@@ -99,7 +99,7 @@ func (c *Connector) ReadBinding(ctx context.Context, username string) (result bo
 
 func (c *Connector) CheckDatabaseExists(ctx context.Context, dbName string) (result bool, err error) {
 	return result, c.withDefaultDBConnection(func(db *sql.DB) (err error) {
-		statement := `SELECT 1 FROM master.dbo.sysdatabases where name=@p1`
+		statement := `SELECT 1 FROM sys.databases where name=@p1`
```
Comment (fnaranjo-vmw) [inline providers/terraform-provider-csbsqlserver/connector/connector.go:102]: Thanks!

**Pull request merged**